test/foundry/AqueductV1Pair.t.sol

```diff
@@ -49,5 +50,169 @@ contract AqueductV1PairIntegrationTest is AqueductTester {
         // Assert
         uint256 totalSupply = aqueductV1Pair.totalSupply();
         assertEq(totalSupply, expectedLiquidity);
+
+        uint256 balanceOf = aqueductV1Pair.balanceOf(ADMIN);
+        uint256 expectedBalance = expectedLiquidity - aqueductV1Pair.MINIMUM_LIQUIDITY();
+        assertEq(balanceOf, expectedBalance);
+
+        uint256 tokenABalanceOf = superTokenA.balanceOf(address(aqueductV1Pair));
+        assertEq(tokenABalanceOf, INIT_SUPER_TOKEN_BALANCE);
+
+        uint256 tokenBBalanceOf = superTokenB.balanceOf(address(aqueductV1Pair));
+        assertEq(tokenBBalanceOf, INIT_SUPER_TOKEN_BALANCE);
+
+        (uint112 _reserve0, uint112 _reserve1, uint32 _blockTimestampLast) = aqueductV1Pair.getReserves();
+        assertEq(_reserve0, INIT_SUPER_TOKEN_BALANCE);
+        assertEq(_reserve1, INIT_SUPER_TOKEN_BALANCE);
+        assertEq(_blockTimestampLast, block.timestamp);
+    }
+
+    function test_mint_position_and_provide_liquidity_BROKEN() public {
+        uint256 expectedInitialLiquidity = 10 * 10 ** 18;
+
+        {
+            uint256 superTokenAAmount = 10 * 10 ** 18;
+            uint256 superTokenBAmount = 10 * 10 ** 18;
+
+            vm.startPrank(ADMIN);
+            addLiquidity(superTokenA, superTokenB, aqueductV1Pair, ADMIN, superTokenAAmount, superTokenBAmount);
+            vm.stopPrank();
+
+            uint256 totalSupply = aqueductV1Pair.totalSupply();
+            assertEq(totalSupply, expectedInitialLiquidity);
+
+            (uint112 _reserve0, uint112 _reserve1, uint time) = aqueductV1Pair.getRealTimeReserves();
+            assertEq(_reserve0, superTokenAAmount);
+            assertEq(_reserve1, superTokenBAmount);
+            assertEq(time, block.timestamp);
+        }
+
+        int96 flowRate = 1000000000;
+
+        vm.startPrank(ADMIN);
+        sf.host.callAgreement(
+            sf.cfa,
+            abi.encodeWithSelector(
+                sf.cfa.createFlow.selector,
+                superTokenA,
+                address(aqueductV1Pair),
+                flowRate,
+                new bytes(0) // placeholder - always pass in bytes(0)
+            ),
+            "0x" //userData
+        );
+        vm.stopPrank();
+
+        vm.warp(block.timestamp + 60);
+
+        uint32 latestTime = uint32(block.timestamp);
+        uint32 nextBlockTime = latestTime + 10;
+
+        uint256 lpToken0Amount = 1 * 10 ** 18;
+        // FIXME: tuple values are other way round to typescript test??
+        (uint112 _reserve0New, uint112 _reserve1New) = aqueductV1Pair.getReservesAtTime(nextBlockTime);
```

Comment:
this code returns:

```sh
  _reserve0:                9999999930210000487
  _reserve1:                10000000070000000000
```

whereas the same code in the hardhat tests returns:

```sh
_reserve0:                 10000000071000000000
_reserve1:                 9999999929213000501
```

When I was running this test earlier, I was getting identical values but reversed (pretty sure my code was doing the same thing in both cases). The values I've pulled from above differ slightly but still look reversed. You can test this out by only running the foundry tests and `mint:dynamic_reserves`. You should see the results in the console. Run forge test with `-vvv`.

Can discuss in standup tomorrow if non of this makes sense. Mainly raising this as a PR now to get your thoughts

Reply:
I believe this is because the factory sorts the addresses in ascending order when creating a pair. So for example, if you have tokenA=0x2... and tokenB=0x1..., then the pool will actually be initialized such that the first token is tokenB and the second token is tokenA. 

In our tests, we can either account for this logic, or maybe it would be simpler to force tokenA's address to always be lesser than tokenB's using vm.etch().

Reply:
Good shout. 

Messed around for a bit with `vm.etch` and couldn't get it to work. Was able to point an address at the right bytecode but when calling a function on the token, it would just revert. 

Ended up swapping the tuple round and left a fixme comment for now